Ryujinx.Ava/Ui/ViewModels/ControllerSettingsViewModel.cs

```diff
@@ -0,0 +1,907 @@
+            }
+        }
+
+        private void HandleOnGamepadDisconnected(string id)
+        {
+            Dispatcher.UIThread.Post(() =>
+            {
+                LoadDevices();
+            });
+        }
+
+        private void HandleOnGamepadConnected(string id)
+        {
+            Dispatcher.UIThread.Post(() =>
+            {
+                LoadDevices();
+            });
+        }
+
+        private string GetCurrentGamepadId()
+        {
+            if (_device < 0)
+            {
+                return string.Empty;
+            }
+
+            string selected = Devices[Device].Id;
+
+            if (selected == null || selected == "disabled")
+            {
+                return null;
+            }
+
+            return selected.Split("/")[1].Split(" ")[0];
+        }
+
+        public void LoadControllers()
+        {
+            Controllers.Clear();
+
+            if (_playerId == PlayerIndex.Handheld)
+            {
+                Controllers.Add(new(ControllerType.Handheld, LocaleManager.Instance["ControllerSettingsControllerTypeHandheld"]));
+
+                Controller = 0;
+            }
+            else
+            {
+                Controllers.Add(new(ControllerType.ProController, LocaleManager.Instance["ControllerSettingsControllerTypeProController"]));
+                Controllers.Add(new(ControllerType.JoyconPair, LocaleManager.Instance["ControllerSettingsControllerTypeJoyConPair"]));
+                Controllers.Add(new(ControllerType.JoyconLeft, LocaleManager.Instance["ControllerSettingsControllerTypeJoyConLeft"]));
+                Controllers.Add(new(ControllerType.JoyconRight, LocaleManager.Instance["ControllerSettingsControllerTypeJoyConRight"]));
+
+                if (Config != null && Controllers.ToList().FindIndex(x => x.Type == Config.ControllerType) != -1)
+                {
+                    Controller = Controllers.ToList().FindIndex(x => x.Type == Config.ControllerType);
+                }
+                else
+                {
+                    Controller = 0;
+                }
+            }
+        }
+
+        private static string GetShrinkedGamepadName(string str)
+        {
+            const string ShrinkChars = "...";
+            const int MaxSize = 50;
+
+            if (str.Length > MaxSize)
+            {
+                return str.Substring(0, MaxSize - ShrinkChars.Length) + ShrinkChars;
+            }
+
+            return str;
+        }
+
+        public void LoadDevices()
+        {
+            lock (Devices)
+            {
+                Devices.Clear();
+                DeviceList.Clear();
+                Devices.Add(("disabled", LocaleManager.Instance["ControllerSettingsDeviceDisabled"]));
+
+                foreach (string id in _mainWindow.InputManager.KeyboardDriver.GamepadsIds)
+                {
+                    IGamepad gamepad = _mainWindow.InputManager.KeyboardDriver.GetGamepad(id);
```

Comment:
```suggestion
                    using IGamepad gamepad = _mainWindow.InputManager.KeyboardDriver.GetGamepad(id);
```
Then you can remove the dispose call.